src/util/dom.ts

```diff
@@ -0,0 +1,25 @@
+import { Elem } from '../types.js'
+
+const template = document.createElement('template')
+
+const createElem = (elem: Elem): HTMLElement => {
+  if (elem instanceof HTMLElement) return elem
+  template.replaceChildren()
+  template.insertAdjacentHTML('afterbegin', elem)
+  return template.firstElementChild as HTMLElement
+}
```

Comment:
```suggestion
const createTemplateEl = () => document.createElement('template')

const createElem = (elem: Elem): HTMLElement => {
  if (elem instanceof HTMLElement) return elem
  const template = createTemplateEl()
  template.insertAdjacentHTML('afterbegin', elem)
  return template.firstElementChild as HTMLElement
}
```
함수내부에서 외부 변수를 직접조작하는게 좋지는 않은것 같은데 replaceChildren으로 하는 것이 더 나은 장점이 있나요??
(한편으로 같은 module내 변수를 조작하고 가까이 붙어있어서 사소한 사항일수도 있는거 같네요)

Reply:
module 내부에서만 존재하는 함수 및 변수이니 클로저와 비슷한 개념으로 생각하시면 될 것 같습니다. 
template 변수는 dom.ts 내부에서만 존재하면서 createElem이 활용하는 자유변수입니다.

template이라는 HTMLElement는 말그대로 '템플릿'으로서만 사용되는 녀석이라서 
굳이 함수 호출시마다 매번 새로 생성해줄 필요가 없다고 생각했어요. 

`document.createElement('template')` vs. `template.replaceChildren()` 둘 중 어느쪽이 비용이 더 드는지는 모르겠지만
형남님이 제안하신 방식이든 제 방식이든 성능상 큰 차이는 없을 것 같긴 해요 :)

Reply:
유의미한 차이가 아니면 수정할 필요가 없다고 생각합니다:)